autoload/sj/elm.vim

```diff
@@ -0,0 +1,247 @@
+" vim: foldmethod=marker
+
+" Split / Join {{{1
+"
+" function! sj#elm#Join() {{{2
+"
+" Attempts to join the multiline tuple, record or list closest
+" to current curspor position.
+function! sj#elm#Join()
+  let [from, to] = sj#elm#ClosestMultilineBraces()
+
+  if from[0] < 0
+    return 0
+  endif
+
+  let original = sj#GetByPosition([0, from[0], from[1]], [0, to[0], to[1]])
+  let joined = sj#elm#JoinOuterBraces(sj#elm#JoinLines(original))
+  call sj#ReplaceByPosition([0, from[0], from[1]], [0, to[0], to[1]], joined)
+
+  return 1
+endfunction
+
+" function! sj#elm#Split() {{{2
+"
+" Attempts to split the tuple, record or list fursest to current
+" curspor position on the same line.
+function! sj#elm#Split()
+  call sj#PushCursor()
+
+  let [from, to] = sj#elm#CurrentLineOutermostBraces(col('.'))
+
+  if from < 0
+    return 0
+  endif
+
+  let parts = sj#elm#SplitParts(from, to)
+
+  if len(parts) <= 2
+    return 0
+  endif
+
+  let replacement = join(parts, "\n")
+
+  call cursor(line('.'), from)
+  let [previousLine, previousCol] = searchpos('\S', 'bn')
+  if previousLine == line('.') && previousCol > 0 && sj#elm#CharAt(previousCol) =~ '[=:]'
+    let replacement = "\n".replacement
+    let from = previousCol + 1
+  end
+
+  call sj#ReplaceCols(from, to, replacement)
+
+  call sj#PopCursor()
+
+  return 1
+endfunction
```

Comment:
In the `return 0` branches, you don't pop the pushed cursor. This isn't a huge deal, but if there was an error elsewhere, the stored cursor might kick in on some extra pop. Ideally, I'd keep the pushes and pops perfectly balanced.

The way I usually manage this is with a try-finally block. Something like:

``` vim
try
  call sj#PushCursor()

  " do stuff
  if problem
    return 0
  endif

  " finish doing stuff
  return 1
finally
  call sj#PopCursor()
endtry
```

That way, any return would trigger the `finally` clause.

Reply:
Great, that's exactly what I needed to clean these up :) thanks a bunch.